ddl/generated_column.go

```diff
@@ -65,6 +84,34 @@ func checkDependedColExist(dependCols map[string]struct{}, cols []*table.Column)
 	return nil
 }
 
+// findPositionRelativeColumn return a pos relative to added generated column position
+func findPositionRelativeColumn(cols []*table.Column, pos *ast.ColumnPosition) (int, error) {
+	position := len(cols)
+	// get column position default is append behind
+	// For "alter table ... add column(...)", the position will be nil.
+	// For "alter table ... add column ... ", the position will be default one.
+	if pos == nil {
+		return position, nil
+	}
+	if pos.Tp == ast.ColumnPositionFirst {
+		position = 0
+	} else if pos.Tp == ast.ColumnPositionAfter {
+		var col *table.Column
+		for _, c := range cols {
+			if c.Name.L == pos.RelativeColumn.Name.L {
+				col = c
+				break
+			}
+		}
+		if col == nil {
+			return -1, ErrBadField.GenWithStackByArgs(pos.RelativeColumn, "generated column function")
+		}
+		// Insert position is after the mentioned column.
```

Comment:
`position` cannot be after a `column`, you can say 'Inserted column is after the mentioned column'.

Reply:
Yes. Shouldn't be a verb here.